### PR TITLE
Automate release process

### PR DESCRIPTION
**Pull request by allieychen**
- Create cloudbuild_release.yaml file to cover the release processes.
- Update the release doc.

Tested: created such trigger in private project and tested the release process.
Issue: [issue 170](https://github.com/googlegenomics/gcp-variant-transforms/issues/170)

### CODE REVIEW

File: cloudbuild_release.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# - Pushes that image to Container Registry.
+# - Runs all integration test from that very image.
+# - Pushes the verified image to the main project (gcp-variant-transforms).
+
+steps:
+  - name: 'gcr.io/cloud-builders/docker'
+    args:
+      - 'build'
+      - '--tag=gcr.io/${PROJECT_ID}/gcp-variant-transforms:${COMMIT_SHA}'
+      - '--file=docker/Dockerfile'
```
Comment (kemp-google) [inline cloudbuild_release.yaml]: args: [ 'build', '-t', 'gcr.io/${PROJECT_ID}/gcp-variant-transforms:${COMMIT_SHA}', 'docker/' ]

you can break it out on lines if you want, but the `-t` and dropping `--file` should help make it more concise

File: cloudbuild_release.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      - '.'
+    id: 'build-gcp-variant-transforms-docker'
+  - name: 'gcr.io/cloud-builders/docker'
```
Comment (kemp-google) [inline cloudbuild_release.yaml]: Can you add a newline between steps?  I find it increases readability significantly

File: cloudbuild_release.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    args:
+      - 'push'
+      - 'gcr.io/${PROJECT_ID}/gcp-variant-transforms:${COMMIT_SHA}'
+    id: 'push-gcp-variant-transforms-docker'
+  - name: 'gcr.io/${PROJECT_ID}/gcp-variant-transforms:${COMMIT_SHA}'
+    args:
+      - 'bash'
+      - './deploy_and_run_tests.sh'
+      - '--keep_image'
```
Comment (kemp-google) [inline cloudbuild_release.yaml]: Why keep_image?

File: cloudbuild_release.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      - '--skip_build'
+      - '--project ${PROJECT_ID}'
+      - '--image_tag ${COMMIT_SHA}'
+      - '--run_unit_tests'
+      - '--run_preprocessor_tests'
+      - '--run_bq_to_vcf_tests'
+      - '--run_all_tests'
+      - '--test_name_prefix cloud-ci-'
+    id: 'test-gcp-variant-transforms-docker'
+  - name: 'gcr.io/cloud-builders/docker'
```
Comment (kemp-google) [inline cloudbuild_release.yaml]: You shouldn't need this step: we will be running it in the gcp-variant-transforms project, and if we aren't we shouldn't cross tag

File: cloudbuild_release.yaml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    args:
+      - 'tag'
+      - 'gcr.io/${PROJECT_ID}/gcp-variant-transforms:${COMMIT_SHA}'
+      - 'gcr.io/gcp-variant-transforms/gcp-variant-transforms:${COMMIT_SHA}'
+    id: 'tag-gcp-variant-transforms-docker'
+images:
```
Comment (kemp-google) [inline cloudbuild_release.yaml]: Nor should you need this AFAIK since you've already pushed

**Comment by allieychen** [comment]
Thanks Aaron! One thing to clarify, what I am thinking is that the build trigger will be added in `gcp-variant-transforms-test`, so all tests are actually running in `gcp-variant-transforms-test`, which I think it would be more convenient since we have the quotas there (also, accessible to some of the test files). If it succeeds, then we push the image to `gcp-variant-transforms`.

File: cloudbuild_release.yaml (hunk 4 above)
Comment (kemp-google) [reply cloudbuild_release.yaml]: OK you've convinced me to leave this as is :)

Let's change the step ids to something like

build-image
deploy-to-test-project
test-image
deploy-to-serving-project

so it's at least clear the difference?

**Pull request merged**